tests/unittest/pywbem/test_subscriptionmanager.py

```diff
@@ -369,3 +371,341 @@ def test_create_not_owned_subscription(self):
             for filter_ in sub_mgr.get_all_filters(server_id):
                 sub_mgr.remove_filter(server_id, filter_.path)
             assert self.get_object_count(sub_mgr, server_id) == 0
+
+
+def setup_mock_environment():
+    """
+    Setup a pywbem mock environment suitable for running subscription tests.
+    This is the same definition as the setup in the unittests above.
+    """
+    skip_if_moftab_regenerated()
+
+    server = WbemServerMock(interop_ns='interop')
+    # pylint: disable=attribute-defined-outside-init
+    conn = server.wbem_server.conn
+    classnames = ['CIM_IndicationSubscription',
+                  'CIM_ListenerDestinationCIMXML',
+                  'CIM_IndicationFilter', ]
+
+    schema = DMTFCIMSchema(DMTF_TEST_SCHEMA_VER, TESTSUITE_SCHEMA_DIR,
+                           verbose=False)
+
+    conn.compile_schema_classes(classnames, schema.schema_pragma_file,
+                                verbose=False)
+    return conn
+
+
+OK = True
+RUN = True
+FAIL = False
+
+
+TESTCASES_SUBMGR = [
+
+    # Testcases for Subscription Tests multiple valid and invalid options
+    # in __init__, add_server, add_filter, add_destinations
+
+    # Each list item is a testcase tuple with these items:
+    # * desc: Short testcase description.
+    # * kwargs: Keyword arguments for the test function:
+    #   * submgr_id: submgr id
+    #   * connection_attrs: WBEMConnection attributes. Ehis variable casues
+    #     an add_server to be executed. If it is a list, multiple
+    #     a server is added for each dict in the list.  This value uses the
+    #     keyword 'MOCKSERVER' to initiated the pywbem_mock environment instead
+    #     of a WBEMConnection.
+    #   * filter_attrs: dictionary of attributes for add_filter. If it is
+    #     a list, one filter is added for each dict in the list
+    #   * dest_attrs: dictionary of attributes for add_listener_destinations.
+    #     Since add_listener_destination can add lists, there is no option
+    #     to add a list for this item.
+    #   * subscription_attrs: dictionary of attributes for add_subscription    #
```

Comment:
The approach to specify the method arguments does not work well for subscriptions, which take instance paths as input. Maybe specifying the filter and destination instances via index into the list of filter_attrs / dest_attrs would work, at least for the good cases.